From a9ea65912eb12e6670eec71ca4828dd445ceedc9 Mon Sep 17 00:00:00 2001
From: Donn Seeley <donn.seeley@windriver.com>
Date: Tue, 12 Jun 2012 10:41:43 -0600
Subject: [PATCH 11/19] emgd: Fix uninitialized variables.

The compiler exposed two parallel instances of a variable that could be
used before set, causing two functions to potentially return a garbage
value in the situation where they had no work to do.

Signed-off-by: Donn Seeley <donn.seeley@windriver.com>
---
 .../emgd/pvr/services4/srvkm/common/deviceclass.c  |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/emgd/pvr/services4/srvkm/common/deviceclass.c b/drivers/gpu/drm/emgd/pvr/services4/srvkm/common/deviceclass.c
index 2867a08..7b6e9cb 100644
--- a/drivers/gpu/drm/emgd/pvr/services4/srvkm/common/deviceclass.c
+++ b/drivers/gpu/drm/emgd/pvr/services4/srvkm/common/deviceclass.c
@@ -1325,7 +1325,7 @@ PVRSRV_ERROR PVRSRVSwapToDCBufferKM(IMG_HANDLE	hDeviceKM,
 									IMG_UINT32	ui32ClipRectCount,
 									IMG_RECT	*psClipRect)
 {
-	PVRSRV_ERROR eError;
+	PVRSRV_ERROR eError = PVRSRV_OK;
 	PVRSRV_DISPLAYCLASS_INFO *psDCInfo;
 	PVRSRV_DC_BUFFER *psBuffer;
 	PVRSRV_QUEUE_INFO *psQueue;
@@ -1488,7 +1488,7 @@ IMG_EXPORT
 PVRSRV_ERROR PVRSRVSwapToDCSystemKM(IMG_HANDLE	hDeviceKM,
 									IMG_HANDLE	hSwapChainRef)
 {
-	PVRSRV_ERROR eError;
+	PVRSRV_ERROR eError = PVRSRV_OK;
 	PVRSRV_QUEUE_INFO *psQueue;
 	PVRSRV_DISPLAYCLASS_INFO *psDCInfo;
 	PVRSRV_DC_SWAPCHAIN *psSwapChain;
-- 
1.7.9.7

